src/clr-addons/datagrid/datagrid-state-persistence/column-hidden-state-persistence.directive.ts

```diff
@@ -10,6 +10,8 @@ import { takeUntil } from 'rxjs/operators';
   selector: '[clrDgHideableColumn]',
 })
 export class ColumnHiddenStatePersistenceDirective implements OnInit, OnDestroy {
+  @Input()
+  clrDgHideableColumn?: { hidden?: boolean; key?: string };
```

Comment:
I'm not that happy to extend clrDgHideableColumn as this is a clarity directive, not clarity-addons.
I would suggest to set the fieldKey with a new directive, which should be placed to clr-dg-column, so we may use it for other purposes too.
So basically you can extend "DatagridFieldDirective" with a different selector to not match the clarity one and provide a getter which returns either the one or the other and use this getter here instead of "this.columnDirective.clrDgField"

Reply:
Thank you for your feedback. I also had this concern a bit and changed it now according to your suggestion. `DatagridFieldDirective` has now a second selector and provides a getter for the "persistence" key.

Also updated the documentation a bit.

Reply:
this looks good now thanks!
maybe just a small hint for future development in github (and other git repositories with merge/pull requests): just add new commits to the development branch.
because if you use amend and force push, we as reviewers can't see what the latest changes since the last review were and the discussions don't have reference points anymore

Reply:
Yes sure, I will avoid ammending commits :)